### PR TITLE
chore(storybook): migrate dev-novel to storybook

### LINKED ISSUE #3051: Move from DevNovel to Storybook
In the next major release Storybook (v4 still in alpha) will support [HTML application](https://github.com/storybooks/storybook/blob/master/docs/src/pages/basics/guide-html/index.md). It can be a good opportunity to move our showcase from [DevNovel](https://github.com/algolia/dev-novel) to Storybook to leverage the community & plugins of Storybook. It also brings consistency across our different flavours since React & Vue are already on Storybook. 

Related https://github.com/algolia/angular-instantsearch/issues/226

### PR DESCRIPTION
**Pull request by francoischalifour**
This migrates our stories from `dev-novel` to `storybook`.

The global environment variable `__DEV__` respects the storybook environment (set to `false` when deployed and `true` when developing).

Link to storybook: https://deploy-preview-3376--instantsearchjs.netlify.com/stories

Closes #3051

---

⚠️ This will need to be deployed (therefore merged) at the same time as the corresponding documentation deployment after having updated the story links in the doc.

### CODE REVIEW

File: .storybook/decorators.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  rightPanelPlaygroundElement.appendChild(hits);
+
+  search.addWidget(
+    instantsearch.widgets.hits({
+      container: hits,
+      templates: {
+        item: `
+<div
+  class="hits-image"
+  style="background-image: url({{image}})"
```
Comment (tkrugg) [inline .storybook/decorators.js]: I think you need quotes around the image?
`background-image: url("{{image}}")`

**Comment by francoischalifour** [comment]
Documentation website is deploying algolia/doc#2506. Going to merge this PR in about 15 minutes.

**Comment by samouss** [comment]
The are some examples that are broken: 

- Geo examples since it's not the same dataset
- The highlighted array does not work anymore (we can remove it since we don't have a proper dataset)

**Comment by francoischalifour** [comment]
Highlight with array has never worked, we can remove it.

Do you mind updating the GeoSearch stories?

**Pull request merged**